### PR TITLE
fix issue 640

### PR DESCRIPTION
**Pull request by zurgl**
Should fix the issue describe [here](https://github.com/near/near-api-js/issues/640)

### CODE REVIEW

File: lib/near.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Near {
         this.connection = connection_1.Connection.fromConfig({
             networkId: config.networkId,
             provider: { type: 'JsonRpcProvider', args: { url: config.nodeUrl } },
-            signer: config.signer || { type: 'InMemorySigner', keyStore: config.keyStore || config.deps.keyStore }
+            signer: config.signer || { type: 'InMemorySigner', keyStore: config.keyStore || config.deps?.keyStore }
```
Comment (volovyk-s) [inline lib/near.js:34]: @zurgl please, add changes to the `.ts` files. Then run `yarn build`. `near.js` is a build artifact.
Also, can you add a test that is failing without this fix? Thanks!

**Comment by zurgl** [comment]
@volovyk-s I've added the path to `near.ts` and a test case.

btw, I'm just patching a deprecated functionality.

**Comment by volovyk-s** [comment]
> @volovyk-s I've added the path to `near.ts` and a test case.
> 
> btw, I'm just patching a deprecated functionality.

Yeah, you are right. Developers can face this issue only if they are following an old API.
We will completely delte `deps` object soon, so I would say that we should just close this issue and PR and follow the new API's.
But anyway, thank you for pointing this out! We have lot's of other work to do ;-)

**Comment by tiendq** [comment]
> 

I'm sorry but where is your new API? I'm following [official docs](https://docs.near.org/docs/api/naj-quick-reference) and got this error too.

`TypeError: Cannot read property 'keyStore' of undefined`

**Comment by volovyk-s** [comment]
Thanks @zurgl , let's continue in https://github.com/near/near-api-js/issues/740

**Comment by volovyk-s** [comment]
@zurgl reopening this one, the fix is correct, my bad.
Thanks, @tiendq !

**Review by volovyk-s — APPROVED** [review]

**Review by ailisp — APPROVED** [review]
Looks good. Thank you @zurgl ! Could you please resolve the conflict so we can merge.

**Comment by volovyk-s** [comment]
Copied to https://github.com/near/near-api-js/pull/752 since I can not merge this PR because of conflicts.

**Pull request closed without merging**